src/providers/ogr/qgsogrconnpool.h

```diff
@@ -61,7 +61,13 @@ class QgsOgrConnPoolGroup : public QObject, public QgsConnectionPoolGroup<QgsOgr
     Q_OBJECT
 
   public:
-    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ) { initTimer( this ); }
+    QgsOgrConnPoolGroup( QString name ) : QgsConnectionPoolGroup<QgsOgrConn*>( name ), mRefCount( 0 ) { initTimer( this ); }
+    void ref() { ++mRefCount; }
```

Comment:
Uhm, so `Q_ASSERT( mRefCount > 0 );` in `unref` is the only one which can be tested unless I'm misunderstanding what you mean.